static/js/message_edit.js

```diff
@@ -455,7 +455,7 @@ function edit_message($row, raw_content) {
             is_editable,
             is_content_editable,
             is_widget_message: is_widget_message(message),
-            has_been_editable: editability !== editability_types.NO,
+            has_been_editable: editability === editability_types.FULL,
```

Comment:
Since we have changed this to be true only when editability is `FULL`, the `is_widget_message` condition [here](https://github.com/zulip/zulip/blob/8fc811dfa92a7212bcf7e807f95603ea82fe19bd/static/templates/message_edit_form.hbs#L73) will not occur anytime and we would not show this tooltp for widget message. Not sure whether we want to show that widgets cannot be edited elsewhere.

Also, we should fix [this line](https://github.com/zulip/zulip/blob/8fc811dfa92a7212bcf7e807f95603ea82fe19bd/static/js/message_edit.js#L549) since `message-edit-timer` element is only present when `has_been_editable` is true (i.e. editability is `FULL`).

Reply:
So should I remove `is_widget_message ` ?

Reply:
If it is not used anywhere in the `message_edit_form` template, which I guess should be correct, then it can be removed for now. We may add it later if we want to mention somewhere that editing is not allowed due to it being a widget message.